src/metabase/query_processor/pivot.clj

```diff
@@ -529,10 +538,11 @@
      (qp.setup/with-qp-setup [query query]
        (let [rff               (or rff qp.reducible/default-rff)
              query             (lib/query (qp.store/metadata-provider) query)
-             pivot-options     (or
-                                (not-empty (select-keys query [:pivot-rows :pivot-cols]))
-                                (pivot-options query (get-in query [:info :visualization-settings])))
-             query             (assoc-in query [:middleware :pivot-options] pivot-options)
-             all-queries       (generate-queries query pivot-options)
+             pivot-opts        (or
+                                (pivot-options query (get query :viz-settings))
+                                (pivot-options query (get-in query [:info :visualization-settings]))
+                                (not-empty (select-keys query [:pivot-rows :pivot-cols :pivot-measures])))
```

Comment:
Is it worth adding tests to make sure we follow the precedence order for these 3 pivot-opts sources?

e.g. make sure `(not-empty (select-keys query [:pivot-rows :pivot-cols :pivot-measures]))` beats `(get-in query [:info :visualization-settings])` etc